src/ui/uiMain.ts

```diff
@@ -23,27 +17,20 @@ export class UiMain {
   }
 
   private async startup(): Promise<void> {
-    // Say hi and ask the main process for new channels.
+    // Init communications.
     this.ipcUiService.mainLog('Started.');
-    this.ipcUiService.onUiMessage = async (m) => {
-      return this.onUiMessage(m);
-    };
     this.ipcUiService.mainPing();
     this.ipcUiService.start();
-  }
 
-  private async onUiMessage(message: IpcUiMessage): Promise<IpcUiResponse> {
-    const messageType = message.type;
-    switch (messageType) {
-      case IpcUiMessageType.UTILS_PING: {
-        const msgPing: IpcUiMessageOf<IpcUiMessageType.UTILS_PING> = message;
-        this.ipcUiService.mainLog(`PING: ${msgPing.data.message}`);
-        return null;
-      }
-      default:
-        this.ipcUiService.mainLog(`Unhandled ui message: ${message}`);
-        return assertUnreachable(messageType);
-    }
+    // Say Hi.
+    const resp: IpcWorkerResponseOf<IpcWorkerMessageType.UTILS_PING> =
```

Comment:
We couldn't send mesasges during startup before, because the worke channel wasn't initiated yet.

This is no longer an issue. The messages will just wait until there is a channel before they are sent.